2-api/server.js

```diff
@@ -127,7 +146,94 @@ app.post("/availability", (req, res) => {
 
 })
 
+// create new order
+app.post("/customers/:customerId/orders", (req, res) => {
+
+    const { customerId } = req.params;
+    const { order_date, order_reference } = req.body;
+    let query = `SELECT * FROM customers WHERE id = $1`
+
+    pool.query(query, [customerId])
+        .then(result => {
+            if (result.rows.length) {
+                pool.query(`INSERT INTO orders(order_date,order_reference,customer_Id) VALUES ($1, $2, $3)`,
+                    [order_date, order_reference, customerId])
+                    .then(() => res.send("order added"))
+                    .catch(error => {
+                        return res.status(500).send({ msg: "failed", err: error.detail })
+                    })
+            } else {
+                return res.status(400).send({ error: "id doesnt exists" })
+            }
+        })
+
+})
+
+// update an existing customer (name, address, city and country)
+app.put("/customers/:customerId", (req, res) => {
+    const { customerId } = req.params;
+    const { name, address, city, country } = req.body;
 
+    pool.query(`SELECT * FROM customers WHERE id = $1`, [customerId])
+        .then(result => {
+            if (result.rows.length) {
+                console.log(result.rows)
+                pool.query(`UPDATE customers SET name= $1, address=$2, city=$3, country=$4 WHERE id= $5`, [name, address, city, country, customerId])
```

Comment:
What if we want to update only one field? currently this will overwrite everything for the customer, even if they are not provided => will be set to null

Hint: use [Coalesce](https://www.postgresqltutorial.com/postgresql-coalesce/) to update only if provided and in case null then it takes the next value specified